# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../libmusicbrainz/gcc43.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -ruN libmusicbrainz-2.1.5-orig/lib/c_wrapper.cpp libmusicbrainz-2.1.5/lib/c_wrapper.cpp
--- libmusicbrainz-2.1.5-orig/lib/c_wrapper.cpp	2006-11-14 17:34:49.000000000 +0100
+++ libmusicbrainz-2.1.5/lib/c_wrapper.cpp	2010-08-03 12:31:50.284171857 +0200
@@ -24,6 +24,7 @@
 #include "musicbrainz.h"
 #include "trm.h"
 #include "mb_c.h"
+#include <cstring>
 
 extern "C"
 {
diff -ruN libmusicbrainz-2.1.5-orig/lib/comhttpsocket.cpp libmusicbrainz-2.1.5/lib/comhttpsocket.cpp
--- libmusicbrainz-2.1.5-orig/lib/comhttpsocket.cpp	2006-11-14 17:34:49.000000000 +0100
+++ libmusicbrainz-2.1.5/lib/comhttpsocket.cpp	2010-08-03 12:31:50.284171857 +0200
@@ -20,6 +20,7 @@
 #endif
 #include <stdlib.h>
 #include <assert.h>
+#include <cstring>
 
 const char* g_strCOMVer = "libmusicbrainz/"VERSION;
 
diff -ruN libmusicbrainz-2.1.5-orig/lib/comsocket.cpp libmusicbrainz-2.1.5/lib/comsocket.cpp
--- libmusicbrainz-2.1.5-orig/lib/comsocket.cpp	2006-11-14 17:34:49.000000000 +0100
+++ libmusicbrainz-2.1.5/lib/comsocket.cpp	2010-08-03 12:31:50.284171857 +0200
@@ -37,6 +37,7 @@
 #include <netinet/tcp.h>
 #include <errno.h>
 #include <stdio.h>
+#include <cstring>
 #include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
diff -ruN libmusicbrainz-2.1.5-orig/lib/http.cpp libmusicbrainz-2.1.5/lib/http.cpp
--- libmusicbrainz-2.1.5-orig/lib/http.cpp	2007-05-11 08:21:21.000000000 +0200
+++ libmusicbrainz-2.1.5/lib/http.cpp	2010-08-03 12:31:50.284171857 +0200
@@ -32,6 +32,8 @@
 #endif
 #include <fcntl.h>    
 #include <errno.h>    
+#include <cstring>
+#include <cstdlib>
 
 #ifdef WIN32
 #include <winsock.h>
diff -ruN libmusicbrainz-2.1.5-orig/lib/sigclient.cpp libmusicbrainz-2.1.5/lib/sigclient.cpp
--- libmusicbrainz-2.1.5-orig/lib/sigclient.cpp	2006-11-14 17:34:49.000000000 +0100
+++ libmusicbrainz-2.1.5/lib/sigclient.cpp	2010-08-03 12:31:50.288172077 +0200
@@ -33,6 +33,8 @@
 #pragma warning(disable:4786)
 #endif
 
+#include <cstring>
+
 #include "sigclient.h"
 #include "comhttpsocket.h"
 
